olive/passes/olive_pass.py

```diff
@@ -120,12 +120,65 @@ def generate_search_space(
 
         # Get the config class with default value or default search value
         config_class, default_config = cls.get_config_class(accelerator_spec, disable_search)
+
+        if not disable_search:
+            # Replace user-provided values with Categorical if user intended to search
+            config = cls.identify_search_values(config, default_config)
+
         # Generate the search space by using both default value and default search value and user provided config
         config = validate_config(config, config_class)
 
         config = cls._resolve_config(config, default_config)
         return cls._init_fixed_and_search_params(config, default_config)
 
+    @classmethod
+    def identify_search_values(
+        cls,
+        config: Dict[str, Any],
+        default_config: Dict[str, PassConfigParam],
+    ):
+        """Conditionally, replace user provided search values with Categorical."""
+        for name, param in default_config.items():
+            if param.search_defaults and name in config:
+                value = config[name]
+
+                # Creat a dummy config class to validate the value
+                dummy_value_config = {name: (param.type_, None)}
+                dummy_value_model = create_model(
+                    f"SearchableParamConfig_{name}_value", **dummy_value_config, __base__=BaseModel
+                )
+
+                # Validate if the user provided a "non-searchable" value
```

Comment:
I don't this this is needed? We are only interest in the case where the user provides a value that is list. for the rest, line 129 can validate the input. 

I think the logic can be simplified to:
```
if not isinstance(value, list) and list:
    continue 

try:
164-